polyfill/test/parseResults.js

```diff
@@ -52,6 +52,11 @@ stdin.on('end', function () {
   const unexpectedPasses = [];
   for (const test of tests) {
     const { result, scenario, file } = test;
+    if (result.message === 'Debugger attached.\nWaiting for the debugger to disconnect...\n') {
+      // work around https://github.com/nodejs/node/issues/34799 when running tests in the debugger
+      result.message = '';
+      result.pass = true;
```

Comment:
If the engine prints this, and then crashes immediately, will we claim the test passes? If so, could you check if that's avoidable? Maybe this workaround could go upstream in test262-harness? (Feel free to argue that the debugging improvement is more important than this potential issue.)

Reply:
My understanding is that `message` contains all console output from the test. If there's no output, then the test passed. If the test code failed or crashed (but Node caught the failure), then there will be a non-blank message. So I think (correct me if this is a bad assumption) that it's not possible for a failing or crashing test to produce that message. 

Note that the message is actually a "sandwich", where the `Debugger attached.\n` output happens before any test code executes, and `Waiting for the debugger to disconnect...\n` is emitted when the test process ends. So if somehow Node itself were to abruptly crash before the test exits, then the second message would not be output and the code inside the `if` statement above would not run.

Does this address your concerns?

It also could make sense to put the workaround upstream (see https://github.com/bterlson/test262-harness/issues/143 that I just filed), but IMHO that shouldn't delay putting it here. If it's accepted upstream, then this condition would become a no-op and could be removed here without harming anything.

Reply:
> `Waiting for the debugger to disconnect...\n` is emitted when the test process ends. So if somehow Node itself were to abruptly crash before the test exits, then the second message would not be output

Okay, that resolves my paranoia. Thanks!